### PR TITLE
CI: Filter any SSLSocket ResouceWarnings from dependencies

### LINKED ISSUE #44732: CI: Flaky ResourceWarning: unclosedssl.SSLSocket
https://github.com/pandas-dev/pandas/runs/4399631223?check_suite_focus=true

```
AssertionError: Caused unexpected warning(s): [('ResourceWarning', ResourceWarning("unclosed <ssl.SSLSocket fd=8, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.1.0.137', 51532), raddr=('52.216.164.101', 443)>"), '', 1043)]
```

Initial guess: it may be coming from one of our dependencies like https://github.com/boto/boto3/issues/454

Potential solution:
https://github.com/boto/botocore/pull/1810#issuecomment-749961250
https://github.com/mintel/pytest-localstack/issues/44#issuecomment-848258701

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #44732
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

### CODE REVIEW

File: pandas/_testing/_warnings.py [hunk 1]
```diff
@@ -147,9 +147,15 @@ def _assert_caught_no_extra_warnings(
...
-                actual_warning.message
+            # GH 44732: Don't make the CI flaky by filtering SSL-related
+            # ResourceWarning from dependencies
+            unclosed_ssl = (
+                "unclosed transport <asyncio.sslproto._SSLProtocolTransport",
+                "unclosed <ssl.SSLSocket",
+            )
+            if (
+                actual_warning.category == ResourceWarning
+                and str(actual_warning.message) in unclosed_ssl
```
Comment (jbrockmendel) [inline pandas/_testing/_warnings.py:158]: should be `any(str(actual_warning.message) in x for x in unclosed_ssl)`?

Comment (mroeschke) [reply pandas/_testing/_warnings.py]: Thanks. Ya the original check was faulty.

**Comment by jreback** [comment]
thanks @mroeschke

**Comment by jbrockmendel** [comment]
We should be able to get rid of [this](https://github.com/pandas-dev/pandas/blob/c58d8ee6e93fc1ba7aa4a40dc71488cc9807ed43/pandas/tests/io/parser/common/test_chunksize.py#L210) xfail now right ?

**Comment by mroeschke** [comment]
> We should be able to get rid of [this](https://github.com/pandas-dev/pandas/blob/c58d8ee6e93fc1ba7aa4a40dc71488cc9807ed43/pandas/tests/io/parser/common/test_chunksize.py#L210) xfail now right ?

Yup should be good to remove

**Pull request merged**